netspeed/src/backend.c

```diff
@@ -207,6 +207,43 @@ get_ip_address_list (const char *iface_name,
     return list;
 }
 
+GSList*
+get_ip6_address_list (const char *iface_name)
+{
+    GSList *list = NULL;
+    struct ifaddrs *ifaces;
+
+    if (getifaddrs (&ifaces) != -1) {
+        char ip6[INET6_ADDRSTRLEN];
+
+        for (struct ifaddrs *iface = ifaces; iface != NULL; iface = iface->ifa_next) {
+            if (iface->ifa_addr == NULL)
+                continue;
+
+            if (!g_strcmp0 (iface->ifa_name, iface_name))
+            {
+                void *sinx_addr = NULL;
+
+                if (iface->ifa_addr->sa_family == AF_INET6) {
```

Comment:
you can combine these 2 clauses in one if statement, i.e.:
`if (!g_strcmp0 (iface->ifa_name, iface_name)) && (iface->ifa_addr->sa_family == AF_INET6))`

Reply:
or place family check before comparing the interface names like NULL check.